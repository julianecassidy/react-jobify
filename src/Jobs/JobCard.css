.JobCard{
    background-color: white;
    width:60%;
    margin-left:20%;
    border-radius: 10px;
    padding:0px 8px 8px 8px;
    margin-bottom:20px;
    box-shadow: 10px 10px 10px #18215A;
    color:black;
}
