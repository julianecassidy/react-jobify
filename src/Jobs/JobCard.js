import { convertAndFormat } from "../helperFuncs";
import "./JobCard.css";
/** Component for JObCard
 *  Renders a card with job details
 * 
 *  props: job {jobData}
 * 
 *  state: none
 *  
 * links to: none
 * 
 * JobCardList -> JobCard
 */

function JobCard({ job }) {
    console.debug("JobCard");

    const salary = convertAndFormat(job.salary);

    return (
        <div className="JobCard">
            <h2>{job.title}</h2>
            <h3>{job.companyName}</h3>

            <div>
                {"salary" in job && job.salary > 0 && (
                    <div>
                        Salary: 
                        ${salary}
                    </div>
                )}

                <div>
                    {"equity" in job && Number(job.equity) > 0 && (
                        <div>
                            Equity: {job.equity}
                        </div>
                    )}
                </div>
            </div>

        </div>
    );
}

export default JobCard;