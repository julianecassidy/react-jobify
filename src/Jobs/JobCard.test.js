import { render } from "@testing-library/react";
import JobCard from "./JobCard";

const testJob1 = {
    title: "Test",
    companyName: "Smith and Co",
    companyHandle: "smith",
    salary: 30000,
    equity: 0.3
};

const testJob2 = {
    title: "Other Test",
    companyName: "Smith and Co",
    companyHandle: "smith",
    salary: null,
    equity: 0
};

test("renders without crashing", function () {
    render(<JobCard job={testJob1} />)
})

test("matches snapshot", function () {
    const { container } = render(<JobCard job={testJob1} />);
    expect(container).toMatchSnapshot();
})

test("job with all fields renders correctly", function () {
    const { container } = render(<JobCard job={testJob1} />);
    const jobCard = container.querySelector(".JobCard")
    
    expect(jobCard).toContainHTML("<h2>Test</h2>");
    expect(jobCard).toContainHTML("<h3>Smith and Co</h3>");
    expect(jobCard).toContainHTML("Salary: $30,000");
    expect(jobCard).toContainHTML("Equity: 0.3");
});

test("job with missing salary and equity renders correctly", function () {
    const { container } = render(<JobCard job={testJob2} />);
    const jobCard = container.querySelector(".JobCard")
    
    expect(jobCard).toContainHTML("<h2>Other Test</h2>");
    expect(jobCard).toContainHTML("<h3>Smith and Co</h3>");
    expect(jobCard).not.toContainHTML("Salary:");
    expect(jobCard).not.toContainHTML("Equity:");
});