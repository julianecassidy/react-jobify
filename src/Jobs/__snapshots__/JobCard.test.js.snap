// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="JobCard"
  >
    <h2>
      Test
    </h2>
    <h3>
      Smith and Co
    </h3>
    <div>
      <div>
        Salary: $
        30,000
      </div>
      <div>
        <div>
          Equity: 
          0.3
        </div>
      </div>
    </div>
  </div>
</div>
`;
